daemon/cmd/daemon.go

```diff
@@ -474,9 +474,15 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 	// Propagate identity allocator down to packages which themselves do not
 	// have types to which we can add an allocator member.
 	//
+	// **NOTE** The identity allocator is not yet initialized; that happens
+	// below. We've only allocated the structure at this point.
+	//
 	// TODO: convert these package level variables to types for easier unit
 	// testing in the future.
-	ipcache.IdentityAllocator = d.identityAllocator
+	ipcache.IPIdentityCache = ipcache.NewIPCache().
```

Comment:
I generally like this API, but it also leaves these methods available for external usages **outside** of construction. Is that desired here? It may be. 

However, if these methods are **only** meant for construction, then the functional options pattern maybe nicer, where those methods would be typed "options" instead of generally available structure methods.

Reply:
The more I look at it, the more I think that the ipcacheConfig{...} approach at least gives less of an impression that the fields are optional. With the structure there we can also document that the fields are all required, so then there's a required config option with documentation that says that each of the fields is required (+ semantics / how they fit into ipcache). So maybe I'll look into that.

Reply:
Sounds good, I didnt know there was a particular scenario you were trying to mitigate against. If that's the case I think that's good enough reason to take this approach, long with your comments about daemon being the only place which inits it.